services/go-filler/log-fetcher/internal/listener/listener.go

```diff
@@ -0,0 +1,106 @@
+package listener
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"math/big"
+	"sync"
+
+	"github.com/base-org/RIP-7755-poc/services/go-filler/bindings"
+	"github.com/base-org/RIP-7755-poc/services/go-filler/log-fetcher/internal/chains"
+	"github.com/base-org/RIP-7755-poc/services/go-filler/log-fetcher/internal/clients"
+	"github.com/base-org/RIP-7755-poc/services/go-filler/log-fetcher/internal/config"
+	"github.com/base-org/RIP-7755-poc/services/go-filler/log-fetcher/internal/handler"
+	"github.com/base-org/RIP-7755-poc/services/go-filler/log-fetcher/internal/store"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+type Listener interface {
+	Start(ctx context.Context) error
+	Stop()
+}
+
+type listener struct {
+	outbox  *bindings.RIP7755Outbox
+	handler handler.Handler
+	logs    chan *bindings.RIP7755OutboxCrossChainCallRequested
+	stop    chan struct{}
+	wg      sync.WaitGroup
+}
+
+func NewListener(srcChainId *big.Int, cfg *config.Config, queue store.Queue) (Listener, error) {
+	srcChain, err := chains.GetChainConfig(srcChainId, cfg.RPCs)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	h, err := handler.NewHandler(cfg, srcChain, queue)
+	if err != nil {
+		return nil, err
+	}
+
+	contractAddress := srcChain.Contracts.Outbox
+	if contractAddress == common.HexToAddress("") {
+		return nil, fmt.Errorf("source chain %s missing Outbox contract address", srcChain.ChainId)
+	}
+
+	client, err := clients.GetEthClient(srcChain)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get eth client: %v", err)
+	}
+	outbox, err := bindings.NewRIP7755Outbox(contractAddress, client)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create Outbox contract binding: %v", err)
+	}
+
+	return &listener{
+		outbox:  outbox,
+		handler: h,
+		logs:    make(chan *bindings.RIP7755OutboxCrossChainCallRequested),
+		stop:    make(chan struct{}),
+	}, nil
+}
+
+func (l *listener) Start(ctx context.Context) error {
```

Comment:
We need to somehow track which block we've processed, so we can always start at the most recent unprocessed block. Perhaps writing this metadata to the datastore would be best.